pkg/controllermanager/controller/shoot/state/finalizer/reconciler.go

```diff
@@ -0,0 +1,90 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package finalizer
+
+import (
+	"context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	controllermanagerconfigv1alpha1 "github.com/gardener/gardener/pkg/controllermanager/apis/config/v1alpha1"
+	"github.com/gardener/gardener/pkg/controllerutils"
+)
+
+// FinalizerName is the finalizer used by the controller.
+const FinalizerName = "core.gardener.cloud/shootstate"
+
+// Reconciler reconciles ShootState objects and ensures the finalizer
+// exists during "Migrate" and "Restore" phases of Shoot migration to
+// another Seed.
+type Reconciler struct {
+	// Client is the API Server client used by the Reconciler.
+	Client client.Client
+	Config controllermanagerconfigv1alpha1.ShootStateControllerConfiguration
+}
+
+// Reconcile reconciles ShootStates.
+func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	log := logf.FromContext(ctx)
+
+	shootState := &gardencorev1beta1.ShootState{}
+	if err := r.Client.Get(ctx, request.NamespacedName, shootState); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.V(1).Info("Did not manage to retrieve ShootState object")
```

Comment:
for consistencies with other controllers:
```suggestion
			log.V(1).Info("Object is gone, stop reconciling")
```

Reply:
Thanks for pointing this message mismatch, fixed it in my latest updates.